src/qtmaximise.h

```diff
@@ -0,0 +1,35 @@
+// Copyright 2015, Christopher J. Foster and the other displaz contributors.
+// Use of this code is governed by the BSD-style license found in LICENSE.txt
+
+#ifndef QT_MAXIMISE_H_INCLUDED
+#define QT_MAXIMISE_H_INCLUDED
+
+#include <QAction>
+#include <QMainWindow>
+
+class QActionMaximise : public QAction
+{
+    Q_OBJECT
+
+    public:
+        QActionMaximise(QMainWindow* mainWindow = 0) : QAction(tr("Maximise"),mainWindow), m_mainWindow(mainWindow)
+        {
+            setShortcut(Qt::Key_F11);
+            connect(this, SIGNAL(triggered(bool)), this, SLOT(handleTriggered(bool)));
+        }
+
+    public slots:
+        void handleTriggered(bool checked)
+        {
+            const bool maximised = m_mainWindow->windowState()&Qt::WindowMaximized;
+            if (maximised)
+                m_mainWindow->showNormal();
+            else
+                m_mainWindow->showMaximized();
+        }
```

Comment:
I'm entirely happy to use American spelling for consistency.

It's true that this is specific to `QMainWindow`, fair enough.  My underlying quibble is that creating a new class for this seems like a lot of boilerplate for a tiny piece of functionality.  The alternative is much less code:

Two additional lines in `PointViewerMainWindow::PointViewerMainWindow`:
```C++
QAction* maximizeAct = viewMenu->addAction("Maximize");
maximizeAct->setShortcut(Qt::Key_F11);
connect(maximizeAct, SIGNAL(triggered(bool)), this, SLOT(toggleMaximized()));
```

Seven lines to define the new slot, plus one in the header.
```C++
PointViewerMainWindow::toggleMaximized()
{
    if (windowState() & Qt::WindowMaximized)
        showNormal();
    else
        showMaximized();
}
```

So a total of 10 new lines for this functionality, vs 30 lines and a new header file.  What you've got is better factored, but I'm not really convinced the tradeoff is worth it.